cirq/sim/sparse_simulator.py

```diff
@@ -232,6 +251,39 @@ def _base_iterator(
             )
             sim_state.log_of_measurement_results.clear()
 
+    def simulate_expectation_values_sweep(
+        self,
+        program: 'cirq.Circuit',
+        observables: Dict[str, 'cirq.PauliSum'],
+        params: 'study.Sweepable',
+        qubit_order: ops.QubitOrderOrList = ops.QubitOrder.DEFAULT,
+        initial_state: Any = None,
+        permit_terminal_measurements: bool = False,
+    ) -> List[Dict[str, float]]:
+        if not permit_terminal_measurements and program.are_any_measurements_terminal():
+            raise ValueError(
+                'Provided circuit has terminal measurements, which may '
+                'skew expectation values. If this is intentional, set '
+                f'permit_terminal_measurements=True.\n\nCircuit:\n{program}'
```

Comment:
Circuits can be very big. I'd recommend just printing the terminal measurements (moment, qubit). 
```
terminal measurements: 
 - Moment: 3,  qubits: "(0,1)", "(0,2)"
 - Moment: 5, qubits: "(5,6)"
```

Reply:
This information isn't returned by `are_any_matches_terminal`. If it isn't strictly necessary, I'd prefer to just drop the circuit from the error message.

Reply:
SGTM

Reply:
Done.